[17.0][FIX] sale_blanket_order: filter default taxes by company for all users

StefanRijnhart opened the pull request:
**Before this commit**
    
When adding a product on a sale blanket order line, all the taxes of the product from all accessible companies were added to the blanket order line.
    
**After this commit**
    
When adding a product on a sale blanket order line, only the taxes of the product from the company on the blanket order are added to the blanket order line. This mimics the treatment of taxes on sale order lines when adding a new product.

Also, some field and view attributes were updated in analogy to sale.order.line.

StefanRijnhart commented:
Tests fixed in https://github.com/OCA/sale-workflow/pull/3290

rousseldenis reviewed (approved):
LGTM

espo-tony reviewed (approved):
:+1: LGTM

CasVissers-360ERP reviewed (approved):
Functional review

StefanRijnhart commented:
@OCA/crm-sales-marketing-maintainers ready to merge, I think.

ivs-cetmix reviewed (changes requested):
Hi @StefanRijnhart , thank you for your contribution!

Could you please improve your commit message for better changes traceability?
Eg like this:

```
[FIX] sale_blanket_order: filter default taxes by company for all users

**Before this commit**

What was the flow before 

**After this commit**

How it works now

StefanRijnhart committed:
[FIX] sale_blanket_order: filter default taxes by company for all users

**Before this commit**

When adding a product on a sale blanket order line, all the taxes of the product from all accessible companies were added to the blanket order line.

**After this commit**

When adding a product on a sale blanket order line, only the taxes of the product from the company on the blanket order are added to the blanket order line. This mimics the treatment of taxes on sale order lines when adding a new product.

Also, some field and view attributes were updated in analogy to sale.order.line.

ivs-cetmix reviewed (approved):
@StefanRijnhart thank you for your contribution!

/ocabot merge patch

StefanRijnhart commented:
Note to self: commit already ended up in https://github.com/OCA/sale-blanket/pull/8, so no backport required.
